### PR TITLE
feat(price-scale-options): introduce missing `PriceScaleOptions` based on Version 4.1 TV Docs

### PR DESCRIPTION
**Pull request by CodexLink**
# Intro

I stumbled upon this library and was able to explore it for a while since I'm interested in using this library. Later on, I found out that some options or properties are not embedded in the `abstract.price_scale`, so I added them to this PR.

## Main Changes

The following PR adds other missing properties defined in the [TV Docs (Ver. 4.1)](https://tradingview.github.io/lightweight-charts/docs/api/interfaces/PriceScaleOptions) `interface PriceScaleOptions`

## Other Changes

There are some changes that I presented in this PR, and those are the following:
- docs
- - `Chart` consistency in the Sidebar
- - `abstract.py`'s `price_scale` function signature to docs.
- files
- - `.gitignore` -> Building the docs is getting included in the `git` so I created a `.gitignore`
- - `price_scale_options.py` -> Demo
- - `abstract.py` -> I added `Optional` to some parameters in `abstract.price_scale`.

G3PUhNnHkz

### About the `example` file
Please let me know where to put it appropriately because the existing examples that I saw are correlated with the docs and I have no plans on creating documentation about it, **yet.**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+docs/build/**
+docs/source/_ext/**
```
Comment (louisnw01) [inline .gitignore:2]: _ext contains style formatting; should this be in here?

File: docs/source/reference/charts.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Charts
+# `Charts`
```
Comment (louisnw01) [inline docs/source/reference/charts.md:1]: "Charts" is not an object so does not need back ticks surrounding it.

File: examples/7_price_scale_options/price_scale_options.py
Comment (louisnw01) [inline examples/7_price_scale_options/price_scale_options.py:1]: I Don't think an example is neccessary for this

**Comment by louisnw01** [comment]
Hey @CodexLink,

All looks good; if you could review my comments that would be great, and then ill get it merged into main.

Louis

File: .gitignore (hunk 1 above)
Comment (CodexLink) [reply .gitignore]: Oh, my bad on this one, I will remove the entry `docs/source/_ext/**` as I overlooked it. 😅

File: docs/source/reference/charts.md (hunk 2 above)
Comment (CodexLink) [reply docs/source/reference/charts.md]: Okay, will remove back ticks.

File: examples/7_price_scale_options/price_scale_options.py
Comment (CodexLink) [reply examples/7_price_scale_options/price_scale_options.py:1]: Understandable since we can explore the Lightweight Charts and tinker from it. Removed.

**Review by CodexLink — COMMENTED** [review]
All changes are applied. Ready for merge or review if needed. Thanks!

File: docs/source/reference/index.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ tables
 2. [`Line`](#Line)
 3. [`Histogram`](#Histogram)
 3. [`HorizontalLine`](#HorizontalLine)
-4. [Charts](#charts)
+4. [`Charts`](#charts)
```
Comment (louisnw01) [inline docs/source/reference/index.md:22]: One last change!

Comment (CodexLink) [reply docs/source/reference/index.md]: Sure! Resolved.

**Pull request merged**